insights/client/apps/compliance/__init__.py

```diff
@@ -27,63 +27,82 @@ def __init__(self, config):
 
     def oscap_scan(self):
         self._assert_oscap_rpms_exist()
-        policies = self.get_policies()
-        if not policies:
+        initial_profiles = self.get_initial_profiles()
+        matching_os_profiles = self.get_profiles_matching_os()
+        profiles = self.profile_union_by_ref_id(matching_os_profiles, initial_profiles)
+        if not profiles:
             logger.error("System is not associated with any profiles. Assign profiles using the Compliance web UI.\n")
             exit(constants.sig_kill_bad)
-        for policy in policies:
+        for profile in profiles:
             self.run_scan(
-                policy['attributes']['ref_id'],
-                self.find_scap_policy(policy['attributes']['ref_id']),
-                '/var/tmp/oscap_results-{0}.xml'.format(policy['attributes']['ref_id']),
-                tailoring_file_path=self.download_tailoring_file(policy)
+                profile['attributes']['ref_id'],
+                self.find_scap_policy(profile['attributes']['ref_id']),
+                '/var/tmp/oscap_results-{0}.xml'.format(profile['attributes']['ref_id']),
+                tailoring_file_path=self.download_tailoring_file(profile)
             )
 
         return self.archive.create_tar_file(), COMPLIANCE_CONTENT_TYPE
 
-    def download_tailoring_file(self, policy):
-        if 'tailored' not in policy['attributes'] or policy['attributes']['tailored'] is False:
+    def download_tailoring_file(self, profile):
+        if ('tailored' not in profile['attributes'] or profile['attributes']['tailored'] is False or
+                'os_minor_version' not in profile['attributes'] or profile['attributes']['os_minor_version'] != self.os_minor_version()):
```

Comment:
@vkrizan I believe removing this condition here would allow the client to work as it does today if there are no os_minor_versions defined in a policy's profiles. I will test it out to be sure, but this would allow us to merge this work today without fear of breaking customers, and we can revisit this condition later to turn off tailoring again when a matching os_minor_version is not found.

Reply:
I've now commented out this line and adjusted a test for this scenario.